web_ui/src/shared/components/date-range-picker-small/date-range-picker-small.component.tsx

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2022-2025 Intel Corporation
+// LIMITED EDGE SOFTWARE DISTRIBUTION LICENSE
+
+import { FC, ReactNode, useEffect, useState } from 'react';
+
+import {
+    Content,
+    DateField,
+    Dialog,
+    DialogTrigger,
+    Flex,
+    Heading,
+    RangeCalendar,
+    Tooltip,
+    TooltipTrigger,
+    useDateFormatter,
+} from '@adobe/react-spectrum';
+import { DateValue, getLocalTimeZone } from '@internationalized/date';
+import { RangeValue } from '@react-types/shared';
+import Calendar from '@spectrum-icons/workflow/Calendar';
+import isEmpty from 'lodash/isEmpty';
+import { RangeCalendarProps } from 'react-aria-components';
+
+import { QuietActionButton } from '../quiet-button/quiet-action-button.component';
+
+interface DateRangePickerSmall extends RangeCalendarProps<DateValue> {
+    hasManualEdition?: boolean;
+    headerContent?: ReactNode;
+}
+
+export const DateRangePickerSmall: FC<DateRangePickerSmall> = ({ hasManualEdition, headerContent, ...props }) => {
+    const formatter = useDateFormatter({ dateStyle: 'long' });
+    const [range, setRange] = useState<RangeValue<DateValue>>();
+    const [focusedDate, setFocusedDate] = useState<DateValue | undefined>();
+
+    const rangeText = isEmpty(range)
+        ? ''
+        : formatter.formatRange(range.start.toDate(getLocalTimeZone()), range.end.toDate(getLocalTimeZone()));
+
+    useEffect(() => {
```

Comment:
would it be possible to avoid this `useEffect` by moving props.value to the initialisation of the state on line 33?